### PR TITLE
Implement `std::ops::Add` for AnsiString

### PR DESCRIPTION
**Pull request by rspencer01**
Sometimes when building AnsiStrings as an application it is useful to do it incrementally (particularly when you have formatted sections that are naturally separated).  This PR implements `std::ops::Add` for AnsiString to facilitate this.

### CODE REVIEW

File: src/ansi.rs [hunk 1]
```diff
@@ -339,6 +339,24 @@ impl<'a> From<(&'a str, &'a [usize], Attr)> for AnsiString<'a> {
     }
 }
 
+impl<'a> std::ops::Add for AnsiString<'a> {
```
Comment (LoricAndre) [inline src/ansi.rs:327]: I'm not sure about the mutating part, is it really necessary ?

Comment (rspencer01) [reply src/ansi.rs:327]: Since `std::ops::Add` takes by value not reference I don't see the issue? This way we don't allocate if we don't need to.

**Review by LoricAndre — APPROVED** [review]

**Comment by LoricAndre** [comment]
Could you please rebase or merge main into this branch ?

**Review by LoricAndre — APPROVED** [review]

**Comment by LoricAndre** [comment]
Better late than never !

**Pull request merged**